pkg/manifests/config.go

```diff
@@ -218,9 +220,23 @@ func NewConfig(content io.Reader, tp bool) (*Config, error) {
 	if c.ClusterMonitoringConfiguration.PrometheusK8sConfig.CollectionProfile != FullCollectionProfile && !tp {
 		return nil, errors.Wrap(ErrConfigValidation, "collectionProfiles is a TechPreview feature, to be able to use a profile different from the default (\"full\") please enable TechPreview")
 	}
-	// Validate CollectionProfile field
-	if !slices.Contains(SupportedCollectionProfiles, c.ClusterMonitoringConfiguration.PrometheusK8sConfig.CollectionProfile) {
-		return nil, errors.Wrap(ErrConfigValidation, fmt.Sprintf(`%q is not supported, supported collection profiles are: %q`, c.ClusterMonitoringConfiguration.PrometheusK8sConfig.CollectionProfile, SupportedCollectionProfiles.String()))
+
+	// Validate the configured collection profile iff tech preview is enabled, even if the default profile is set.
+	if tp {
+		didMisconfigureCollectionProfile := false
+		if !slices.Contains(SupportedCollectionProfiles, c.ClusterMonitoringConfiguration.PrometheusK8sConfig.CollectionProfile) {
+			didMisconfigureCollectionProfile = true
+		}
+		for _, profile := range SupportedCollectionProfiles {
+			var v float64 = 0
+			if profile == c.ClusterMonitoringConfiguration.PrometheusK8sConfig.CollectionProfile {
+				v = 1
+			}
+			cmometrics.ConfiguredCollectionProfile.WithLabelValues(string(profile)).Set(v)
+		}
+		if didMisconfigureCollectionProfile {
```

Comment:
(nit) we can get rid of the intermediate variable
```suggestion
		if !slices.Contains(SupportedCollectionProfiles, c.ClusterMonitoringConfiguration.PrometheusK8sConfig.CollectionProfile) {
```